python/tk_framework_desktopserver/server.py

```diff
@@ -36,12 +38,37 @@ def __init__(self, port=None, debug=True, whitelist=None, keys_path=None):
         self._whitelist = whitelist or self._DEFAULT_WHITELIST
         self._debug = debug
 
-        self._logger_root = logging.getLogger("tk-framework-desktopserver")
+        self._logger_root = get_logger()
+
+        twisted = get_logger("twisted")
 
         if self._debug:
-            # This will take the Twisted logging and forward it to Python's logging.
-            self._observer = log.PythonLoggingObserver("tk-framework-desktopserver.twisted")
-            self._observer.start()
+            # When running the server in debug mode, the twisted framework will print out data
+            # exchanged with the client. Unforutnately, there's no way to filter out that
+            # information from since it is being logged at the INFO level just like regular Twisted
+            # logs.
+
+            # Because the package information is sensitive, we will make sure logs don't propagate
+            # to the root handler.
+            twisted.propagate = False
+
+            # Print Twisted logs to the console.
+            handler = logging.StreamHandler()
```

Comment:
This doesn't seem right to me. 

The whole idea with logging is to present a unified architecture of structured message passing so that a receiving client can determine what to do with that. The client (desktop, shell, some other future integration) should decide what to do with this info, log it to stdout, write it to a file, a syslog, a port etc. 

The library should merely dispatch logging information to the listener without making any assumptions about where this goes. Therefore, I don't think a library should ever define a handler like this. Instead, we should ensure that the information is passed in such a way that it is safe.

Our libraries should never echo anything to stdout/stderr. 

We want this potentially sensitive info to show when in debug mode but not anywhere else? 

Check this out: https://docs.python.org/2/howto/logging-cookbook.html#using-loggeradapters-to-impart-contextual-information

I think a better approach here would be to create a special logger instance for twisted that has the necessary business logic to understand when to discard messages and not (e.g. based on debug logging). Then leave all the formatting, output etc to the client side. All that we should do at this level is to basically cull out twisted messages depending on whether the debug flag is set or not.